corehq/apps/hqwebapp/models.py

```diff
@@ -81,6 +87,32 @@ def __new__(cls, category, action, label=None):
         return super(GaTracker, cls).__new__(cls, category, action, label)
 
 
+class UUIDGeneratorMixin(object):
+    """
+    Automatically generates uuids on __init__ if not generated yet.
+
+    To use: Add this mixin to your model as the left-most class being inherited from
+    and list all field names in UUIDS_TO_GENERATE to generate uuids for.
+
+    NOTE: Where possible, a UUIDField should be used instead of this mixin. But
+    this is needed in cases where migrating a char field to a UUIDField is
+    not possible because some of the existing uuids don't match the
+    UUIDField format constraints.
+    """
```

Comment:
so the issue is that some of these fields are referenced elsewhere for linking. for example, the sms couch_id is what is referenced on smsbillables to tie a billable to an sms. 

i've thought about updating all the references to use pk, but in case we ever decide to shard any of these models we'll need to keep linkage with the random uuid and not pk. so for now i just kept all of them generating a uuid.  (i think i can also now give them a more descriptive name than couch_id so it doesn't sound confusing)